packages/test/test-end-to-end-tests/src/test/gc/gcSweepUnreferencePhases.spec.ts

```diff
@@ -163,46 +164,55 @@ describeNoCompat("GC sweep unreference phases", (getTestObjectProvider) => {
 			"Data Store should be in the summary!",
 		);
 
-		// Wait sweep timeout
+		// Stage 3 - Inactive -> SweepReady //
+		// Wait sweep timeout. The sweep ready nodes' ids will be sent in the GC op.
 		await delay(sweepTimeoutMs);
 		mainDataStore._root.set("send", "op2");
 		await provider.ensureSynchronized();
 		const summary3 = await summarizeNow(summarizer);
-		const summaryTree3 = summary3.summaryTree;
+		assert(
+			summary3.summaryTree.tree[gcTreeKey].type === SummaryType.Handle,
+			"The GC tree should be the same as before",
+		);
+
+		// Stage 4 - SweepReady -> Swept //
+		// Wait for the GC op to be processed so that the sweep ready nodes are swept.
+		await provider.ensureSynchronized();
+		const summary4 = await summarizeNow(summarizer);
+		const summaryTree4 = summary4.summaryTree;
 		// GC graph check
-		const gcState3 = getGCStateFromSummary(summaryTree3);
+		const gcState3 = getGCStateFromSummary(summaryTree4);
 		assert(gcState3 !== undefined, "Expected GC state to be generated");
 		assert(
 			!(dataStoreHandle.absolutePath in gcState3.gcNodes),
 			"Data Store should not exist on gc graph",
 		);
 		// GC Tombstone check
-		const tombstoneState3 = getGCTombstoneStateFromSummary(summaryTree3);
+		const tombstoneState3 = getGCTombstoneStateFromSummary(summaryTree4);
 		assert(tombstoneState3 === undefined, "Nothing should be tombstoned");
 		// GC Sweep check
-		const deletedState3 = getGCDeletedStateFromSummary(summaryTree3);
+		const deletedState3 = getGCDeletedStateFromSummary(summaryTree4);
 		assert(deletedState3 !== undefined, "Should have sweep state");
 		assert(deletedState3.includes(dataStoreHandle.absolutePath), "Data Store should be swept");
 		assert(deletedState3.includes(ddsHandle.absolutePath), "DDS should be swept");
```

Comment:
This is failing currently because the deleted list does not contain DDS ids. I am evaluating how to fix this.